src/OSPSuite.Presentation/Services/DataImporter.cs

```diff
@@ -87,43 +87,41 @@ private static MetaDataCategory createMetaDataCategory<T>(string descriptiveName
       public (IReadOnlyList<DataRepository> DataRepositories, ImporterConfiguration Configuration) ImportDataSets(
          IReadOnlyList<MetaDataCategory> metaDataCategories,
          IReadOnlyList<ColumnInfo> columnInfos,
-         DataImporterSettings dataImporterSettings
+         DataImporterSettings dataImporterSettings,
+         string dataFileName
       )
       {
-         var path = _dialogCreator.AskForFileToOpen(PleaseSelectDataFile, ImportFileFilter, Constants.DirectoryKey.OBSERVED_DATA);
-
-         if (string.IsNullOrEmpty(path))
+         if (string.IsNullOrEmpty(dataFileName))
             return (Array.Empty<DataRepository>(), null);
 
          using (var importerModalPresenter = _applicationController.Start<IModalImporterPresenter>())
          {
-            return importerModalPresenter.ImportDataSets(metaDataCategories, columnInfos, dataImporterSettings, path);
+            return importerModalPresenter.ImportDataSets(metaDataCategories, columnInfos, dataImporterSettings, dataFileName);
          }
       }
 
       public IReadOnlyList<DataRepository> ImportFromConfiguration(
          ImporterConfiguration configuration,
          IReadOnlyList<MetaDataCategory> metaDataCategories,
          IReadOnlyList<ColumnInfo> columnInfos,
-         DataImporterSettings dataImporterSettings
+         DataImporterSettings dataImporterSettings,
+         string dataFileName
       )
       {
-         var fileName = _dialogCreator.AskForFileToOpen(OpenFile, ImportFileFilter, Constants.DirectoryKey.OBSERVED_DATA);
-         
-         if (string.IsNullOrEmpty(fileName))
+         if (string.IsNullOrEmpty(dataFileName))
```

Comment:
same as above

Reply:
Good catch, thanks! Indeed the expected behavior is to return an empty list when no valid path (this would be the case for example if the open dialog is to be cancelled).